src/unitxt/fusion.py

```diff
@@ -44,40 +45,59 @@ class FixedFusion(BaseFusion):
     """FixedFusion operator that combines multiple streams into one based on a fixed number of examples per task.
 
     Args:
-        origins: List of SourceOperator objects.
+        origins: Dict of named SourceOperator objects.
         examples_per_task: Number of examples per task. If None, all examples are returned.
         splits: List of splits to include. If None, all splits are included.
     """
 
     max_instances_per_origin: Optional[int] = None
 
+    # flake8: noqa: C901
     def fusion_generator(self, split) -> Generator:
-        for origin in self.origins:
+        for origin_name, origin in self.origins.items():
             multi_stream = origin()
             if split not in multi_stream:
                 continue
             iterator = iter(multi_stream[split])
             if self.max_instances_per_origin is not None:
                 for _ in range(self.max_instances_per_origin):
                     try:
-                        yield next(iterator)
+                        instance = next(iterator)
+                        if "group" in instance:
+                            if "past_groups" in instance:
+                                instance["past_groups"] = (
+                                    instance["group"] + "/" + instance["past_groups"]
+                                )
+                            else:
+                                instance["past_groups"] = instance["group"]
+                        instance["group"] = origin_name
+                        yield instance
                     except StopIteration:
                         break
             else:
-                yield from iterator
+                for instance in iterator:
+                    if "group" in instance:
+                        if "past_groups" in instance:
+                            instance["past_groups"] = (
+                                instance["group"] + "/" + instance["past_groups"]
+                            )
+                        else:
+                            instance["past_groups"] = instance["group"]
+                    instance["group"] = origin_name
```

Comment:
Same here dont think past_groups are necessary

Reply:
Yes, same thing: trying to work here, in order to save work from the consure (e.g. Splitter)

Reply:
Changed here too